### PR TITLE
feat: Finalize initial version with updated data types and documentation.

### PR DESCRIPTION
**Pull request by mrderyk**
## CONTEXT
This PR contains the final changes for the initial release.

## CHANGES
- Add license file
- Add README
- Move contents of index.ts -> client.ts
- Export client and data types from index.ts
- Document types

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -38,15 +38,85 @@ export type ParsedResult = {
...
+  filter?: string;
+
+  // The query to send to the API.
+  // This is the user input.
+  queryValue?: string;
+
+  // The language the summary should be in.
+  language?: SummaryLanguage;
+
+  rerank?: boolean;
```
Comment (mrderyk) [inline src/types.ts:52]: I'd love some help describing these. :)

File: src/types.ts [hunk 2]
```diff
@@ -38,15 +38,85 @@ export type ParsedResult = {
...
+  rerankNumResults?: number;
+  rerankerId?: number;
+  rerankDiversityBias?: number;
+
+  hybridNumWords: number;
```
Comment (mrderyk) [inline src/types.ts]: And these as well. :)

File: README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+<p align="center">
+  <img style="max-width: 100%;" alt="Welcome to Stream-Query-Client" src="images/projectLogo.png"/>
+</p>
+
+# Stream-Query-Client
+
+Stream-Query-Client is the easiest way to use Vectara's streaming query API. Simply provide a query configuration and a callback to instantly receive stream updates.
```
Comment (cjcenizal) [inline README.md]: I think we should clarify that it's the easiest way to use it in JavaScript. The reader might be looking for something to help them with a Python codebase, for example.

File: src/client.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+              store: config.chat?.store ?? false,
+              conversationId: config.chat?.conversationId,
+            },
+          },
+        ],
+        ...(config.rerank
+          ? {
+              rerankingConfig: {
+                rerankerId: config.rerankerId,
+                ...(config.rerankerId === 272725718
```
Comment (cjcenizal) [inline src/client.ts]: Can we extract this reranker ID value into a variable to provide some more semantics to the reader?

File: src/types.ts (hunk 1 above)
Comment (cjcenizal) [reply src/types.ts:52]:
```
// Reranking orders your search results for increased relevance.
rerank?: boolean;
// Specify how many search results to retrieve and rerank.
rerankNumResults?: number;
// Which reranker will be used.
rerankerId?: number;
// Diversity bias ranges from 0 to 1. 0 will optimize for results that are as closely related to the query as possible. 1 will optimize for results that are as diverse as possible.
rerankDiversityBias?: number;
```

File: src/types.ts (hunk 2 above)
Comment (cjcenizal) [reply src/types.ts]: I actually don't know about these!

**Review by cjcenizal — APPROVED** [review]
Code LGTM! Didn't test locally.

**Pull request merged**